### ISSUE TITLE
Senior Data Engineer

### ISSUE THREAD

**Issue by stevendelarwelle** [stevendelarwelle/remotejobs#7456]
Senior Data Engineer<br>
https://ift.tt/2gjS6xD<br>
<p><img src="https://we-work-remotely.imgix.net/logos/0006/6198/logo.gif?ixlib=rails-2.1.3&amp;w=190&amp;min-h=150" alt="Logo.gif?ixlib=rails 2.1"></p>
<p><strong>Headquarters:</strong> New York, NY<br>
<strong>URL:</strong> <a href="http://www.custora.com">www.custora.com</a></p>
<div><img src="https://s3.amazonaws.com/custora-greenhouse-media/developer.png" width="407" height="263"></div>
<div><br></div>
<div>Custora exists to help our customers improve the relationships they have with their own customers. &nbsp;We do this by ingesting data about every interaction a company has with each of their customers and then making predictions using that data about how those customers will behave in the future. &nbsp;Our customers then use these predictions to tailor their communications.</div>
<div><br></div>
<div>Data engineers at Custora work on the pipelines that sit at the core of this architecture. &nbsp;We work to make these pipelines faster, more fault tolerant and to expand their scope.</div>
<div><br></div>
<div>The volume of data is large: we're working with 7 of the top 20 largest retailers in the world (+ many more not in the top 20), and are ingesting data from them both in a regular batch and in near-real time. &nbsp;</div>
<div><br></div>
<div>We've carefully selected the types of data to ingest to favor high signal data, so we care deeply about maintaining the correctness and completeness of the data being ingested as our models (and therefore the output of our product). &nbsp;</div>
<div><br></div>
<div>Your work directly impacts both the predictions we are able to make, and the day to day performance our customers experience when using our product.</div>
<div><br></div>
<div><strong>Getting more specific, you will:</strong></div>
<div><br></div>
<ul>
<li>Design and build complex data pipelines on the Spark platform, ingesting both batch and real time datasets</li>
<li>Work with our data science team to deploy predictive models at scale</li>
<li>Build tools to continuously validate incoming data and proactively identify and communicate data anomalies before they manifest into problems.</li>
</ul>
<div>We&rsquo;re a small team, so you&rsquo;ll be working on (and be able to meaningfully contribute to) high impact projects from your first day.</div>
<div><br></div>
<div><strong>Sure, but what&rsquo;s it really like?</strong></div>
<div><br></div>
<div>Inspired by Basecamp, we work in ~8 week product cycles. &nbsp;First, we work together (engineering + product) to identify the projects we think will have the biggest impact on our company goals. &nbsp;Here&rsquo;s an example of a recent project we conceptualized and delivered over one of these cycles:&nbsp;</div>
<div><br></div>
<ol>
<li><strong>Migrate self-managed Spark cluster to EMR</strong></li>
</ol>
<div>To lower overall cost, and to be able to easily scale to handle bigger datasets and processing volumes, we recently switched from a self managed Spark cluster to Amazon&rsquo;s Elastic Mapreduce service.</div>
<div><br></div>
<div>Our challenge was to move terabytes of data used by our clients while having no downtime. Some initial concerns were the performance on the EMR cluster and the migration process itself because some clients ingest a combination of live data and batch based data. The scale of our data sent meant that we had to switch several clients at a time to EMR.</div>
<div><br></div>
<div>After ensuring that Hive backed by S3 was performant enough, we built tools to move vast amounts of data in parallel, to redirect requests to the correct cluster as clients were being moved, and to validate the data after migration. Along the way, we also had to reshape the data (in terms of partition size) to ensure efficiency of copying and loading into the Hive database.</div>
<div><br></div>
<div><strong>The Stack:</strong></div>
<div><br></div>
<div>While we make use of a wide variety of tools, our primary web stack is ES6/React and Ruby on Rails deployed on AWS. We make extensive use of R for statistical analysis, and our primary data stores are Hive, MySQL, and Redis.</div>
<div><br></div>
<div>
<strong>What</strong> <strong>it&rsquo;s</strong> <strong>like</strong> <strong>to</strong> <strong>work</strong> <strong>here:</strong>
</div>
<div><br></div>
<ul>
<li>On Monday we eat and meet as a team to chat projects and progress.</li>
<li>We&rsquo;re 50 genuinely nice people; 25 men and 25 women. We work together and experiment with how to do things.&nbsp;</li>
<li>We move quickly. You build something and the next day it comes to life. You see and feel an immediate impact with the collective efforts of the team.</li>
<li>We&rsquo;re building a company and a team we love. We&rsquo;re in it for the long run.</li>
<li><a href="https://s3.amazonaws.com/custora-public/150623_Values_1.2.pdf">Read more about what makes us, us here.</a></li>
<li>Find out more <a href="https://www.builtinnyc.com/2018/01/18/techies-land-dream-jobs">here</a> or <a href="https://www.builtinnyc.com/2017/09/21/top-offices-nyc">here</a>.</li>
</ul>
<div>Custora is an equal opportunity employer. We value diversity. We don&rsquo;t discriminate on the basis of race, religion, color, national origin, gender, sexual orientation, marital status, veteran status, disability status, or socioeconomic status.</div>
<div><br></div>
<div><strong>Qualifications:</strong></div>
<div><br></div>
<ul>
<li>5 or more years of experience as a software engineer.</li>
<li>Degree in Computer Science or a deep competency achieved via other means.</li>
<li>Familiarity with Ruby on Rails, AWS, and SQL-based databases.</li>
<li>High standards for code quality and maintainability.</li>
</ul>
<div><strong>Nice to Have's:</strong></div>
<div><br></div>
<ul>
<li>Experience with R, Scala, Spark, and/or Chef.</li>
<li>Consistent record of delivering significant features or building out platforms and services.</li>
<li>Experience working in e-commerce.</li>
</ul>
<div><strong>The perks</strong></div>
<div><br></div>
<ul>
<li>We&rsquo;re a <a href="https://life.saywerk.com/posts/flexcert-spotlight-custora">flexible work</a> environment</li>
<li>Competitive salary and meaningful equity</li>
<li>Health, dental and vision insurance (100% covered)</li>
<li>Free lunch every day, plus free water &mdash; hot and cold!</li>
<li>Unlimited vacation: take as much time as you need (we recommend at least 3 weeks)</li>
<li>Monthly unlimited MetroCard</li>
<li>401k&nbsp;</li>
</ul>
<p><strong>To apply:</strong> <a href="http://www.custora.com/careers">www.custora.com/careers</a> or email <a href="mailto:erin@custora.com">erin@custora.com</a></p>
<br>
<br>
via Remote Jobr Programming Jobs https://ift.tt/1WipnMk<br>
May 29, 2018 at 11:15AM
